docs/references/api/resource_embedding.rst

```diff
@@ -318,10 +320,294 @@ Computed relationships have good performance as their intended design enable `in
+
+.. code-block:: json
+
+  [
+    {
+      "last_name": "Adams",
+      "predecessor": {
+        "last_name": "Washington"
+      }
+    }
+  ]
+
+.. _recursive_o2m_embed:
+
+Recursive One-To-Many
+---------------------
+
+This case works the same way as a normal :ref:`Many-To-One relationship<many-to-one>`.
+It is the default behavior for when you have a table like this one:
+
+.. tabs::
+
+  .. group-tab:: ERD
+
+    .. image:: ../../_static/employees.png
+
+  .. code-tab:: postgresql SQL
+
+    create table employees (
+      id int primary key generated always as identity,
+      first_name text,
+      last_name text,
+      supervisor_id int references employees(id)
+    );
+
+This query will return the Many-To-One embedding, that is, the supervisors with their supervisees:
+
+.. tabs::
+
+  .. code-tab:: http
+
+    GET /employees?select=last_name,supervisees:employees(last_name) HTTP/1.1
+
+  .. code-tab:: bash Curl
+
+    curl "http://localhost:3000/employees?select=last_name,supervisees:employees(last_name)"
+
+.. code-block:: json
+
+  [
+    {
+      "name": "Smith",
+      "supervisees": [
+       { "name": "Davis" }
+      ]
+    },
+    {
+      "name": "Taylor",
+      "supervisees": [
+        { "name": "Johnson" },
+        { "name": "Miller" }
+      ]
+    }
+  ]
+
+.. _recursive_m2o_embed:
+
+Recursive Many-To-One
+----------------------
+
+Let's take the same ``employees`` table from :ref:`recursive_o2m_embed`.
+To get the Many-To-One relationship, that is, the employees with their respective supervisor, you need to create a function like this one:
+
+.. code-block:: postgresql
+
+  create or replace function supervisor(employees) returns setof employees rows 1 as $$
+    select * from employees where id = $1.supervisor_id
+  $$ stable language sql;
+
+Then, the query would be:
+
+.. tabs::
+
+  .. code-tab:: http
+
+    GET /employees?select=last_name,supervisor(last_name)&id=eq.3 HTTP/1.1
+
+  .. code-tab:: bash Curl
+
+    curl "http://localhost:3000/employees?select=last_name,supervisor(last_name)&id=eq.3"
+
+.. code-block:: json
+
+  [
+    {
+      "last_name": "Miller",
+      "supervisor": {
+        "last_name": "Taylor"
+      }
+    }
+  ]
+
+.. _recursive_m2m_embed:
+
+Recursive Many-To-Many
+----------------------
+
+.. tabs::
+
+  .. group-tab:: ERD
+
+    .. image:: ../../_static/users.png
+
+  .. code-tab:: postgresql SQL
+
+    create table users (
+      id int primary key generated always as identity,
+      first_name text,
+      last_name text,
+      username text unique
+    );
+
+    create table subscriptions (
+      subscriber_id int references users(id),
+      subscribed_id int references users(id),
+      type text,
+      primary key (subscriber_id, subscribed_id)
+    );
+
+To get all the subscribers of a user, define the function:
+
+.. code-block:: postgresql
+
+  create or replace function subscribers(users) returns setof users as $$
```

Comment:
Hm, getting a user's subscriptions (with usernames) works OOTB here? I think we need another computed rel?

Reply:
Yeah, another computed rel was needed. I added it now.